open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/vrp/retrieval/flow/VRPConsentRetrievalUtilTest.java

```diff
@@ -0,0 +1,389 @@
+import com.wso2.openbanking.accelerator.consent.extensions.common.ConsentServiceUtil;
+import com.wso2.openbanking.accelerator.consent.extensions.utils.ConsentExtensionTestUtils;
+import com.wso2.openbanking.accelerator.consent.mgt.dao.models.AuthorizationResource;
+import com.wso2.openbanking.accelerator.consent.mgt.dao.models.ConsentResource;
+import com.wso2.openbanking.accelerator.consent.mgt.service.impl.ConsentCoreServiceImpl;
+import net.minidev.json.JSONArray;
+import net.minidev.json.JSONObject;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.Assert;
+import org.testng.IObjectFactory;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.ObjectFactory;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.wso2.openbanking.accelerator.consent.extensions.authorize.vrp.retrieval.flow.ConsentAuthorizeTestConstants.VRP_WITHOUT_DATA;
+import static org.mockito.Mockito.mock;
+import static org.powermock.api.mockito.PowerMockito.doReturn;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+/**
+ * Test class for consentRetrievalUtil.
+ */
+@PowerMockIgnore({"com.wso2.openbanking.accelerator.consent.extensions.common.*", "net.minidev.*",
+        "jdk.internal.reflect.*"})
+@PrepareForTest({OpenBankingConfigParser.class, OpenBankingConfigParser.class, ConsentServiceUtil.class})
+public class VRPConsentRetrievalUtilTest extends PowerMockTestCase {
+
+    @InjectMocks
+    private final DefaultConsentRetrievalStep defaultConsentRetrievalStep = new DefaultConsentRetrievalStep();
+    private final ConsentRetrievalUtil consentRetrievalUtil = new ConsentRetrievalUtil();
+    @Mock
+    private static ConsentData consentDataMock;
+    @Mock
+    private static ConsentResource consentResourceMock;
+    @Mock
+    private static AuthorizationResource authorizationResourceMock;
+    @Mock
+    ConsentCoreServiceImpl consentCoreServiceMock;
+    @Mock
+    OpenBankingConfigParser openBankingConfigParser;
+    @Mock
+    OpenBankingConfigParser openBankingConfigParse;
+    private static Map<String, Object> configMap;
+    ArrayList<AuthorizationResource> authResources;
+
+
+    @BeforeClass
+    public void initClass() {
+        MockitoAnnotations.initMocks(this);
+        consentDataMock = mock(ConsentData.class);
+        consentResourceMock = mock(ConsentResource.class);
+        authorizationResourceMock = mock(AuthorizationResource.class);
+        consentCoreServiceMock = mock(ConsentCoreServiceImpl.class);
+
```

Comment:
Shall we refactor these classes? seems there are additional new lines in test classes.

Reply:
Fixed in [02c291d]